### PR TITLE
[#120] Add checkstyle support

### PR DESCRIPTION
**Pull request by j-lum**
Included a modified version of checkstyle rules based on AB-4.

Build file modified to run checkstyle as part of the default build process.

Checkstyle violations are fixed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,11 +16,6 @@ deploy:
   on:
     branch: master
 
-addons:
-  apt:
-    packages:
-      - oracle-java9-installer
-
```
Comment (pyokagan) [inline .travis.yml]: Seems like this should have been done with #119 ;-) Please move this into a separate PR.

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -383,6 +388,7 @@ private static String executeCommand(String userInputString) {
             return getUsageInfoForAllCommands();
         case COMMAND_EXIT_WORD:
             executeExitProgramRequest();
+            //Fallthrough
```
Comment (pyokagan) [inline src/seedu/addressbook/AddressBook.java]: Space after the `//` please.

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -608,7 +614,7 @@ private static String getUserInput() {
         return inputLine;
     }
 
-   /*
+    /*
```
Comment (pyokagan) [inline src/seedu/addressbook/AddressBook.java]: Is this change intended? Seems wrong to make the comment gutter mis-aligned.

Comment (j-lum) [reply src/seedu/addressbook/AddressBook.java]: Line 617 was one space short of the required indentation level. Going to fix this by aligning the other lines in the comment to match.

File: .travis.yml (hunk 1 above)
Comment (j-lum) [reply .travis.yml]: Reverted it for this branch.

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -206,6 +206,11 @@
      * ====================================================================
      */
 
+    /**
+     * Main entry point of the application.
+     * If a valid path to a storage file is provided as an argument, that file
+     * will be loaded instead of the default.
```
Comment (pyokagan) [inline src/seedu/addressbook/AddressBook.java]: Is it important to note down how the main() function handles command-line arguments here? How would this approach scale to an application which handles many more command-line arguments? Would we have to note them all down in the javadoc? That would be hard to maintain, I would imagine.
